File: src/main/java/org/micoli/php/configuration/schema/valueGenerator/CodeStyleGenerator.java
package org.micoli.php.configuration.schema.valueGenerator;

import com.intellij.psi.codeStyle.CommonCodeStyleSettings;
import java.lang.reflect.Field;
import java.lang.reflect.Modifier;
import java.util.Arrays;
import java.util.List;

public class CodeStyleGenerator implements PropertyValueGenerator {
    List<String> fieldNames = List.of("styleAttribute");

    @Override
    public List<String> getFieldNames() {
        return fieldNames;
    }

    @Override
    public List<String> getValues() {
        Field[] fields = CommonCodeStyleSettings.class.getFields();
        return Arrays.stream(fields)
                .filter(field -> Modifier.isPublic(field.getModifiers()))
                .filter(field -> {
                    Class<?> fieldType = field.getType();
                    return fieldType == int.class
                            || fieldType == Integer.class
                            || fieldType == boolean.class
                            || fieldType == Boolean.class;
                })
                .map(Field::getName)
                .distinct()
                .sorted()
                .toList();
    }
}

File: src/main/kotlin/org/micoli/php/configuration/schema/valueGenerator/CodeStyleGenerator.kt
package org.micoli.php.configuration.schema.valueGenerator

import com.intellij.psi.codeStyle.CommonCodeStyleSettings
import com.intellij.util.containers.stream
import java.lang.reflect.Modifier
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf
import kotlinx.collections.immutable.toImmutableList

class CodeStyleGenerator : PropertyValueGenerator {
    override fun getFieldNames(): ImmutableList<String> = persistentListOf("styleAttribute")

    override fun getValues(): ImmutableList<String> {
        return CommonCodeStyleSettings::class
            .java
            .getFields()
            .stream()
            .filter { it != null }
            .filter { Modifier.isPublic(it.modifiers) }
            .filter {
                it.type == Int::class.javaPrimitiveType ||
                    it.type == Int::class.java ||
                    it.type == Boolean::class.javaPrimitiveType ||
                    it.type == Boolean::class.java
            }
            .map { it.name }
            .distinct()
            .sorted()
            .toList()
            .toImmutableList()
    }
}
